normandy/base/api/relations.py

```diff
@@ -0,0 +1,57 @@
+from django.core.urlresolvers import reverse as django_reverse
+from django.conf import settings
+
+from rest_framework.relations import HyperlinkedRelatedField
+from rest_framework.reverse import preserve_builtin_query_params
+
+
+class CDNHyperlinkedRelatedField(HyperlinkedRelatedField):
+
+    def __init__(self, **kwargs):
+        # super is going to override self.reverse, so stash it and restore it
+        reverse = self.reverse
+        super().__init__(**kwargs)
+        self.reverse = reverse
+
+    def reverse(self, viewname, args=None, kwargs=None, request=None, format=None, **extra):
+        """
+        Specialization of DRF's reverse that uses settings.CDN_URL.
+
+        Does not support DRF API versioning. TODO: Can it?
+        """
+        scheme = getattr(request, 'versioning_scheme', None)
+        if scheme is not None:
+            raise NotImplemented("CDN reverse doesn't support versioning")
+
+        if format is not None:
+            kwargs = kwargs or {}
+            kwargs['format'] = format
+
+        url = django_reverse(viewname, args=args, kwargs=kwargs, **extra)
+        if settings.CDN_URL is None:
+            url = request.build_absolute_uri(url)
+        else:
+            assert url[0] == '/'
+            url = settings.CDN_URL + url[1:]
```

Comment:
[`urllib.parse.urljoin`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) will see the leading `/` in `url` and treat it as a absolute path, stripping out any path components in `CDN_URL`.

That is, if `CDN_URL` is `https://example.com/cdn/`, then `urljoin` will strpi out the `cdn/` part.